homeassistant/components/shelly/__init__.py

```diff
@@ -83,59 +69,111 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
 
     coap_context = await get_coap_context(hass)
 
+    # First check if device is not sleeping
     try:
         async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
             device = await aioshelly.Device.create(
                 aiohttp_client.async_get_clientsession(hass),
                 coap_context,
                 options,
+                True,
             )
     except (asyncio.TimeoutError, OSError) as err:
-        raise ConfigEntryNotReady from err
+        # Not a sleeping device, raise error
+        if not entry.data["sleep_period"]:
+            raise ConfigEntryNotReady from err
+
+    # Sleeping device, make an empty device
+    if device is None:
+        async with async_timeout.timeout(AIOSHELLY_DEVICE_TIMEOUT_SEC):
```

Comment:
Doesn't need timeout if it doesn't make any connection.

Reply:
Removed